internal/metrics/remote_write_test.go

```diff
@@ -0,0 +1,207 @@
+package metrics_test
+
+import (
+	"context"
+	"errors"
+	"os"
+	"time"
+
+	gomock "github.com/golang/mock/gomock"
+	"github.com/golang/snappy"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/project-flotta/flotta-device-worker/internal/metrics"
+	"github.com/prometheus/prometheus/prompb"
+)
+
+var _ = Describe("remote write", func() {
+	var (
+		mockCtrl     *gomock.Controller
+		tsdbInstance *metrics.MockAPI
+		remoteWrite  *metrics.RemoteWrite
+		writeClient  *metrics.MockWriteClient
+		deviceID     = "deviceid"
+	)
+
+	BeforeEach(func() {
+		os.Remove(metrics.LastWriteFileName)
+		mockCtrl = gomock.NewController(GinkgoT())
+		tsdbInstance = metrics.NewMockAPI(mockCtrl)
+		writeClient = metrics.NewMockWriteClient(mockCtrl)
+		remoteWrite = metrics.NewRemoteWrite("", deviceID, tsdbInstance)
+		remoteWrite.Client = writeClient
+	})
+
+	AfterEach(func() {
+		defer GinkgoRecover()
+		mockCtrl.Finish()
+		os.Remove(metrics.LastWriteFileName)
+	})
+
+	It("empty URL feature disabled", func() {
+		remoteWrite.Client = nil
+		remoteWrite.Write()
+	})
+
+	It("new device", func() {
+		tsdbInstance.EXPECT().MaxTime().Return(time.Time{}).Times(1)
+		remoteWrite.Write()
```

Comment:
No `Expect` invocation after `remoteWrite.Write()`. Is it ok?

Reply:
the expect is before. Expect a call to MaxTime.

Reply:
I'll add

Reply:
@gciavarrini done